### PR TITLE
Improving documentation for PAYG integration

### PR DESCRIPTION
**Pull request by 0rnela**
# Description

Detailed and informative comments on bug #1208984 have been added to the documentation.

# Target branches

Which documentation version does this PR apply to?

- [x] Master (Default)
- [x] Manager-4.3 https://github.com/uyuni-project/uyuni-docs/pull/2236
- [x] Manager-4.2 https://github.com/uyuni-project/uyuni-docs/pull/2237

# Links

Fixes https://github.com/SUSE/spacewalk/issues/20709

### CODE REVIEW

**Comment by 0rnela** [comment]
@rjmateus -- as you will see, this is just skeleton text. I had used the content from the [bug 1208984 comment #12](https://bugzilla.suse.com/show_bug.cgi?id=1208984#c12) and tried to rearrange it to the best of my ability for now.

File: modules/installation-and-upgrade/pages/connect-payg.adoc [hunk 1]
```diff
@@ -111,14 +111,55 @@ To delete the SSH connection data to the instance, use <<proc-deleting-connectio
...
+
+[IMPORTANT]
+====
+* The instance needs to have a valid cloud connection registered, otherwise it will not have access to channels.
+
+* If the user removes the cloud packages, the credentials harvesting may stop working.
+
+* Running `registercloudguest --clean` can cause problems if no new cloud connection is registered with the cloud guest command.
+
+* Refreshing or changing the [litera;]``registercloudguest`` connection to the public cloud update infrastructure is fine.  << OM: what does ``fine`` mean?
```
Comment (rjmateus) [inline modules/installation-and-upgrade/pages/connect-payg.adoc:145]:
```suggestion
* Refreshing or changing the [litera;]``registercloudguest`` connection to the public cloud update infrastructure should not interfere with the credentials usage.
```

File: modules/installation-and-upgrade/pages/connect-payg.adoc [hunk 2]
```diff
@@ -111,14 +111,55 @@ To delete the SSH connection data to the instance, use <<proc-deleting-connectio
...
+* files stored in [path]``/etc/zypp/credentials`` 
+* certificate that should alredy exist on the machine (instance?). This certificate is added by the cloud infrastructure.
+* an entry in the machine's [literal]``hosts``file that is needed for the cloud plugin to work. It is automatically set by the installed cloud packages.
+
+With this information available, a zypper plugin is invoked to get a valid authentication header.
+(OM: I need a sentence here to bridge the {payg} connector and the authentication header.)
+
+[IMPORTANT]
+====
+* Removing the zypper credentials or the imported certificate,  it (what is IT - the connector?) will stop working.
```
Comment (rjmateus) [inline modules/installation-and-upgrade/pages/connect-payg.adoc:161]: This section about "== {payg} connector process" I think can be replaced by a warning or suggestion.
 
```suggestion
Any of the following actions/changes in the payg instance will lead to credentials stopping to work:
* Remove zypper credentials files
* Remove the imported certificates
* Remove cloud-specific entries from `/etc/hosts`
* Stop the service named "??"
```

**Review by rjmateus — COMMENTED** [review]
Hey Ornela,
I added some comments. I think we can have just a few alerts

**Comment by 0rnela** [comment]
@rjmateus -- I implemented the changes we talked about. Now I'm just waiting for the service name, and I can ask for approval from other reviewers too.

File: modules/installation-and-upgrade/pages/connect-payg.adoc [hunk 3]
```diff
@@ -111,14 +111,51 @@ To delete the SSH connection data to the instance, use <<proc-deleting-connectio
...
+== Registering {payg} system as a client
+
+You can register a {payg} instance from where you harvest the credentials as a {salt} client.
+The instance needs to have a valid cloud connection registered, otherwise it will not have access to channels.
+If the user removes the cloud packages, the credentials harvesting may stop working.
+
+
+Using [literal]``registercloudguest``::
+* Refreshing or changing the [literal]``registercloudguest`` connection to the public cloud update infrastructure should not interfere with the credentials usage.
+* Running [literal]```registercloudguest --clean`` can cause problems if no new cloud connection is registered with the cloud guest command.
```
Comment (rjmateus) [inline modules/installation-and-upgrade/pages/connect-payg.adoc:136]:
```suggestion
* Running [literal]```registercloudguest --clean`` will cause problems if no new cloud connection is registered with the cloud guest command.
```

**Review by rjmateus — COMMENTED** [review]
a few notes

File: modules/installation-and-upgrade/pages/connect-payg.adoc [hunk 4]
```diff
@@ -111,14 +111,51 @@ To delete the SSH connection data to the instance, use <<proc-deleting-connectio
...
+
+
+== Registering {payg} system as a client
+
+You can register a {payg} instance from where you harvest the credentials as a {salt} client.
+The instance needs to have a valid cloud connection registered, otherwise it will not have access to channels.
+If the user removes the cloud packages, the credentials harvesting may stop working.
+
+
+Using [literal]``registercloudguest``::
```
[the remark this answers is not in the document]
Comment (0rnela) [reply modules/installation-and-upgrade/pages/connect-payg.adoc:134]: Done.

File: .changelog [hunk 5]
```diff
@@ -4,6 +4,8 @@
 # - Fixed error in Bat section of Upgrade Guide (bsc#1234567)
 # For guidelines: https://en.opensuse.org/openSUSE:Creating_a_changes_file_(RPM)
 
+- Improved Pay-as-you-go documentation in the Install/Upgrade Guide
+  (bsc#1208984)
```
Comment (keichwa) [inline .changelog:8]:
```suggestion
- Improved Pay-as-you-go documentation in the Install and Upgrade
  Guide (bsc#1208984)
```

File: modules/installation-and-upgrade/pages/connect-payg.adoc [hunk 6]
```diff
@@ -102,7 +102,7 @@ The following procedure describes how to delete SSH connection data of the insta
 {productname} server uses credentails collected from the instance to connect to the RMT server and to download the packages using reposync.
 These credentials are refreshed every 10 minutes by taskomatic using the defined SSH connection data. Connection to RMT server always uses the last known authentication credentials collected from the {payg} instance.
 
-Up-to date status of the {payg} instance credentials collect is shown in the column Status or on the instance details page.
+Up-to-date status of the {payg} instance credentials collect is shown in the column [literal]``Status`` or on the instance details page.
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/connect-payg.adoc:105]: Wondering whether we can delete the expression "up to date" here?

File: modules/installation-and-upgrade/pages/connect-payg.adoc [hunk 7]
```diff
@@ -111,14 +111,50 @@ To delete the SSH connection data to the instance, use <<proc-deleting-connectio
...
+
+== Registering {payg} system as a client
+
+You can register a {payg} instance from where you harvest the credentials as a {salt} client.
+The instance needs to have a valid cloud connection registered, otherwise it will not have access to channels.
+If the user removes the cloud packages, the credentials harvesting may stop working.
+
+First set up the {payg} instance to collect authentication data, so it can synchronize the channels.
+
+The rest of the process is the same as for any non-public-cloud client and consists of synchronizing channels, automatic bootstrap scriptcreation, activation key creation and starting the registeration.
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/connect-payg.adoc:135]:
```suggestion
The rest of the process is the same as for any non-public-cloud client and consists of synchronizing channels, automatic bootstrap script creation, activation key creation and starting the registration.
```

File: modules/installation-and-upgrade/pages/connect-payg.adoc [hunk 8]
```diff
@@ -111,14 +111,50 @@ To delete the SSH connection data to the instance, use <<proc-deleting-connectio
...
+For more about registering clients, see xref:client-configuration:registration-overview.adoc[].
+
+
+[IMPORTANT]
+====
+Any of the following actions or changes in the {payg} instance will lead to credentials stopping to work:
+
+* removing zypper credentials files
+* removing the imported certificates
+* remove cloud-specific entries from [path]``/etc/hosts``
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/connect-payg.adoc:146]: Why writing "remove" here?

File: modules/installation-and-upgrade/pages/connect-payg.adoc [hunk 9]
```diff
@@ -111,14 +111,50 @@ To delete the SSH connection data to the instance, use <<proc-deleting-connectio
...
+====
+
+
+== Troubleshooting 
+
+Checking the credentils::
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/connect-payg.adoc:152]:
```suggestion
Checking the credentials::
```

File: modules/installation-and-upgrade/pages/connect-payg.adoc [hunk 10]
```diff
@@ -111,14 +111,50 @@ To delete the SSH connection data to the instance, use <<proc-deleting-connectio
...
+* If the script fails to collect the credentials, it should provide an proper error message in the logs and in the {webui}.
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/connect-payg.adoc:153]:
```suggestion
* If the script fails to collect the credentials, it should provide a proper error message in the logs and in the {webui}.
```

**Review by keichwa — CHANGES_REQUESTED** [review]

File: modules/installation-and-upgrade/pages/connect-payg.adoc (hunk 6 above)
Comment (jcayouette) [reply modules/installation-and-upgrade/pages/connect-payg.adoc:105]: "The latest status of the {payg} instance credentials collect is show in....

Comment (0rnela) [inline modules/installation-and-upgrade/pages/connect-payg.adoc:105]:
```suggestion
The status of the {payg} instance credentials collect is shown in the column [literal]``Status`` or on the instance details page.
```

File: modules/installation-and-upgrade/pages/connect-payg.adoc [hunk 11]
```diff
@@ -111,14 +111,50 @@ To delete the SSH connection data to the instance, use <<proc-deleting-connectio
...
+For more about registering clients, see xref:client-configuration:registration-overview.adoc[].
+
+
+[IMPORTANT]
+====
+Any of the following actions or changes in the {payg} instance will lead to credentials stopping to work:
+
+* removing zypper credentials files
+* removing the imported certificates
+* remove cloud-specific entries from [path]``/etc/hosts``
```
Comment (0rnela) [inline modules/installation-and-upgrade/pages/connect-payg.adoc:146]:
```suggestion
* removing cloud-specific entries from [path]``/etc/hosts``
```

**Review by keichwa — APPROVED** [review]
thanks!

**Pull request merged**